x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/LocalModel.java

```diff
@@ -169,4 +188,33 @@ public static void mapFieldsIfNecessary(Map<String, Object> fields, Map<String,
             });
         }
     }
+
+    long acquire() {
+        long count = referenceCount.incrementAndGet();
+        if (count == 1) {
+            trainedModelCircuitBreaker.addEstimateBytesAndMaybeBreak(trainedModelDefinition.ramBytesUsed(), modelId);
+        }
+        return count;
+    }
+
+    public long getReferenceCount() {
+        return referenceCount.get();
+    }
+
+    public long release() {
+        if (referenceCount.decrementAndGet() == 0) {
```

Comment:
should we prevent `release` from ever decrementing below `0`?

Reply:
I added an assertion